### PR TITLE
`Development`: Add new feature tag `Integrated code lifecycle`

### PR DESCRIPTION
**Pull request by krusche**
This PR focused on documentation

### Checklist
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Review Progress
- [x] Code Review 1
- [ ] Code Review 2

## Summary by CodeRabbit

- **Documentation**
  - Updated documentation to reflect the new "Integrated Code Lifecycle" feature.
  - Revised server setup instructions and renumbered feature list.
  - Downgraded certain documentation dependencies for compatibility.

- **Refactor**
  - Renamed references from "Local CI and local VC" to "Integrated Code Lifecycle" across various services to align with new terminology.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,35 +19,36 @@ Artemis brings interactive learning to life with instant, individual feedback on
...
+8**[Grading](https://docs.artemis.cit.tum.de/user/grading/)**: Instructors can configure grade keys for courses and exams to automatically calculate grades and display them to students. Grades can be easily exported as csv files to upload them into university systems (such as Campus online). Instructors can optionally define bonus configurations for final exams to improve student grades according to their grades from a midterm exam or course exercises.
+**[Assessment](https://docs.artemis.cit.tum.de/user/exercises/assessment/)**: Artemis uses double-blind grading and structured grading criteria to improve consistency and fairness. It integrates an assessment training process (based on example submissions and example assessments defined by the instructor), has a grading leader board, and allows students to rate the assessments. Students can complain or ask for more feedback.
```
Comment (reschandreas) [inline README.md:37]:
```suggestion
9. **[Assessment](https://docs.artemis.cit.tum.de/user/exercises/assessment/)**: Artemis uses double-blind grading and structured grading criteria to improve consistency and fairness. It integrates an assessment training process (based on example submissions and example assessments defined by the instructor), has a grading leader board, and allows students to rate the assessments. Students can complain or ask for more feedback.   
```

File: README.md [hunk 2]
```diff
@@ -19,35 +19,36 @@ Artemis brings interactive learning to life with instant, individual feedback on
...
+7. **[Exam mode](https://docs.artemis.cit.tum.de/user/exam_mode/)**: Instructors can create online exams with exercise variants, integrated plagiarism checks, test runs and student reviews. You can find more information on [Exam mode student features](https://artemis.cit.tum.de/features/students) and on [Exam mode instructor features](https://artemis.cit.tum.de/features/instructors).
+8**[Grading](https://docs.artemis.cit.tum.de/user/grading/)**: Instructors can configure grade keys for courses and exams to automatically calculate grades and display them to students. Grades can be easily exported as csv files to upload them into university systems (such as Campus online). Instructors can optionally define bonus configurations for final exams to improve student grades according to their grades from a midterm exam or course exercises.
```
Comment (reschandreas) [inline README.md:36]:
```suggestion
8. **[Grading](https://docs.artemis.cit.tum.de/user/grading/)**: Instructors can configure grade keys for courses and exams to automatically calculate grades and display them to students. Grades can be easily exported as csv files to upload them into university systems (such as Campus online). Instructors can optionally define bonus configurations for final exams to improve student grades according to their grades from a midterm exam or course exercises.
```

**Review by laurenzfb — APPROVED** [review]
documentation changes lgtm

**Review by mateusmm01 — APPROVED** [review]
Changes lgtm 👍

**Review by reschandreas — APPROVED** [review]
lgtm

**Pull request merged**